### PR TITLE
Fix remote machine connection logic

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts [hunk 1]
```diff
@@ -82,6 +84,12 @@ class RemoteMachineTrainingService implements TrainingService {
         restServer.setEnableVersionCheck = this.versionCheck;
         this.log.info('Run remote machine training service.');
         while (!this.stopping) {
+            if (this.sshConnectionPromises.length) {
```
Comment (squirrelsc) [inline src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts:87]: To improve readabilty, and avoid tricks of JS, it's bettter to write like `this.sshConnectionPromises.length > 0`

Comment (squirrelsc) [inline src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts:87]: It looks should be put outside of while loop, as it's one time work.

Comment (SparkSnail) [reply src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts]: fixed

Comment (SparkSnail) [reply src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts]: fixed

**Pull request merged**